src/main/java/programminglife/gui/controller/GraphController.java

```diff
@@ -430,4 +361,27 @@ public void highlightByGenome(int genomeID) {
         oldGenomeList = drawNodeList;
         highlightNodes(drawNodeList, Color.YELLOW);
     }
+
+    /**
+     * Returns the node clicked on else returns null.
+     * @param x position horizontally where clicked
+     * @param y position vertically where clicked
+     * @return nodeClicked {@link DrawableNode} returns null if no note is clicked.
+     */
+    public DrawableNode onClick(double x, double y) {
+        DrawableNode nodeClicked = null;
+        //TODO implement this with a tree instead of iterating.
```

Comment:
This is only for very late in the project since the current version of iterating over all the nodes goes instant already. I have ran this with reasonable size of nodes in the subGraph.